chaincode/fabcar/java/src/main/java/org/hyperledger/fabric/samples/fabcar/CarQueryResult.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.samples.fabcar;
+
+import java.util.Objects;
+
+import org.hyperledger.fabric.contract.annotation.DataType;
+import org.hyperledger.fabric.contract.annotation.Property;
+
+import com.owlike.genson.annotation.JsonProperty;
+
+/**
+ * CarQueryResult structure used for handling result of query
+ *
+ */
+@DataType()
+public final class CarQueryResult {
+	@Property()
+	private final String key;
+	
+	@Property()
+	private final Car record;
+	
+	public CarQueryResult(@JsonProperty("Key") final String key, @JsonProperty("Record") final Car record) {
+		this.key = key;
+		this.record = record;
+	}
+	
+    public String getKey() {
+		return key;
+	}
+
+	public Car getRecord() {
+		return record;
+	}
+
+	@Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if ((obj == null) || (getClass() != obj.getClass())) {
+            return false;
+        }
+
+        CarQueryResult other = (CarQueryResult) obj;
+
+        Boolean recordsAreEquals = this.getRecord().equals(other.getRecord());
+        Boolean keysAreEquals = this.getKey().equals(other.getKey());
+        
+        return recordsAreEquals && keysAreEquals;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.getKey(), this.getRecord());
+    }
+    
+    @Override
+	public String toString() {
+		return "{\"Key\":\"" + key + "\"" +  "\"Record\":{\"" + record + "}\"}";
+	}
```

Comment:
These too.

Reply:
Fixed [in](https://github.com/hyperledger/fabric-samples/pull/149/commits/1ada9a7ca354c9a3dcfe23e1e209ac333ac54dba)